### PR TITLE
Add note that only file rules can be used to validate files

### PR DESCRIPTION
**Pull request by dgvirtual**
Also add explanation regarding rule ``uploaded``, that it makes the file upload required.

### CODE REVIEW

File: user_guide_src/source/libraries/validation.rst [hunk 1]
```diff
@@ -971,6 +971,9 @@ Rule                    Parameter  Description
 ======================= ========== ============================================= ===================================================
 uploaded                Yes         Fails if the name of the parameter does not  ``uploaded[field_name]``
                                     match the name of any uploaded files.
+                                    Prevents form from validating if file upload
+                                    is required.
+
```
Comment (sammyskills) [inline user_guide_src/source/libraries/validation.rst:976]: The description already provides the perfect explanation.

> The validation will fail if the name of the parameter does not match the name of any uploaded files.

Comment (kenjis) [reply user_guide_src/source/libraries/validation.rst:976]: I don't get what the added sentence means.

Comment (kenjis) [reply user_guide_src/source/libraries/validation.rst:976]: How about adding this?

> If you want the file upload to be optional (not required), do not define this rule.

Comment (dgvirtual) [reply user_guide_src/source/libraries/validation.rst:976]: > How about adding this?
> 
> > If you want the file upload to be optional (not required), do not define this rule.

Yep, that is better, thanks.

File: user_guide_src/source/libraries/validation.rst [hunk 2]
```diff
@@ -992,3 +995,7 @@ is_image                Yes         Fails if the file cannot be determined to be
 ======================= ========== ============================================= ===================================================
 
 The file validation rules apply for both single and multiple file uploads.
+
+.. note:: Only rules specifically created for file validation (like the ones listed in the table above) can be used to validate files.
+    Therefore, adding any general rules, like ``permit_empy``, to file validation rules array or string will prevend all rules for the
+    from file from being processed during validation.
```
Comment (kenjis) [inline user_guide_src/source/libraries/validation.rst:1001]: I don't get well "prevend all rules for the from file from being processed during validation."

File: user_guide_src/source/libraries/validation.rst (hunk 1 above)
Comment (dgvirtual) [reply user_guide_src/source/libraries/validation.rst:976]: > The description already provides the perfect explanation.
> 
> > The validation will fail if the name of the parameter does not match the name of any uploaded files.

Technically you must be correct, but there must be a more user-friendly way to say "this rule makes the file required". I read that section and did not get it.

File: user_guide_src/source/libraries/validation.rst (hunk 2 above)
Comment (kenjis) [reply user_guide_src/source/libraries/validation.rst:1001]: How about this?

```suggestion
    Therefore, if you add any general rules, like ``permit_empy`` to file validation rules array or string,
    the file validation will not work correctly.
```

Comment (dgvirtual) [reply user_guide_src/source/libraries/validation.rst:1001]: Well, if you set rules 'permit_empty|max_size[profile_img,1024]|mime_in[profile_img,image/jpeg]', you can uplood 3mb pdf's, and validation will pass.

Comment (dgvirtual) [reply user_guide_src/source/libraries/validation.rst:1001]: Yep, but I cannot believe I messed up my English that much... sorry.

Comment (kenjis) [reply user_guide_src/source/libraries/validation.rst:1001]: ``permit_empty`` (or other general rules) checks `$_POST` array, and the field for the file upload does not exist, so the validation passes.
However, most rules do not permit empty, and `$_POST` has no field for it, so if set, the validation will fail.

I don't know how to explain it briefly.

Comment (dgvirtual) [reply user_guide_src/source/libraries/validation.rst:1001]: Ah, it's complicated. I was also thinking about filing a bug report (it is not good that adding a stray rule disables the other rules), but now I suspect that it is more of a limitation of validation than a bug...

Comment (kenjis) [inline user_guide_src/source/libraries/validation.rst:1001]: How about this?

```suggestion
    Rules for General Use like ``permit_empy`` do not check the uploaded file (in the ``$_FILES`` superglobal),
    therefore, adding these rules to file validation rules array or string will not yield the expected results.
```

File: user_guide_src/source/libraries/validation.rst (hunk 1 above)
Comment (sammyskills) [reply user_guide_src/source/libraries/validation.rst:976]: Yes. https://github.com/codeigniter4/CodeIgniter4/pull/7969#discussion_r1335243837 is better.

File: user_guide_src/source/libraries/validation.rst [hunk 3]
```diff
@@ -992,3 +995,7 @@ is_image                Yes         Fails if the file cannot be determined to be
 ======================= ========== ============================================= ===================================================
 
 The file validation rules apply for both single and multiple file uploads.
+
+.. note:: Only rules specifically created for file validation (like the ones listed in the table above) can be used to validate files.
+    Therefore, adding any general rules, like ``permit_empy``, to file validation rules array or string will prevend all rules for the
```
Comment (sammyskills) [inline user_guide_src/source/libraries/validation.rst:1000]: A typo:
```suggestion
    Therefore, adding any general rules, like ``permit_empty``, to file validation rules array or string will prevend all rules for the
```

**Comment by kenjis** [comment]
```
/github/workspace/user_guide_src/source/libraries/validation.rst:977: ERROR: Malformed table.

Text in column margin in table line 6.

======================= ========== ============================================= ===================================================
Rule                    Parameter  Description                                   Example
======================= ========== ============================================= ===================================================
uploaded                Yes         Fails if the name of the parameter does not  ``uploaded[field_name]``
                                    match the name of any uploaded files.
                                    If you want the file upload to be optional (not required),
                                    do not define this rule.

max_size                Yes         Fails if the uploaded file named in the      ``max_size[field_name,2048]``
looking for now-outdated files... none found
                                    parameter is larger than the second
                                    parameter in kilobytes (kb). Or if the file
                                    is larger than allowed maximum size declared
                                    in php.ini config file -
                                    ``upload_max_filesize`` directive.
max_dims                Yes         Fails if the maximum width and height of an  ``max_dims[field_name,300,150]``
                                    uploaded image exceed values. The first
                                    parameter is the field name. The second is
                                    the width, and the third is the height. Will
                                    also fail if the file cannot be determined
                                    to be an image.
mime_in                 Yes         Fails if the file's mime type is not one     ``mime_in[field_name,image/png,image/jpeg]``
                                    listed in the parameters.
ext_in                  Yes         Fails if the file's extension is not one     ``ext_in[field_name,png,jpg,gif]``
                                    listed in the parameters.
is_image                Yes         Fails if the file cannot be determined to be ``is_image[field_name]``
                                    an image based on the mime type.
======================= ========== ============================================= ===================================================
```
https://github.com/codeigniter4/CodeIgniter4/actions/runs/6407584328/job/17394669739?pr=7969

**Comment by dgvirtual** [comment]
Yep, see that now, will fix later today.

**Review by kenjis — DISMISSED** [review]
Thank you!

File: user_guide_src/source/libraries/validation.rst [hunk 4]
```diff
@@ -992,3 +995,7 @@ is_image                Yes         Fails if the file cannot be determined to be
 ======================= ========== ============================================= ===================================================
 
 The file validation rules apply for both single and multiple file uploads.
+
+.. note:: Only rules specifically created for file validation (like the ones listed in the table above) can be used to validate files.
+    Therefore, adding any general rules, like ``permit_empy``, to file validation rules array or string, the file validation will not
```
Comment (kenjis) [inline user_guide_src/source/libraries/validation.rst:1000]:
```suggestion
    Therefore, adding any general rules, like ``permit_empty``, to file validation rules array or string, the file validation will not
```

**Pull request merged**